app/src/main/java/org/mifos/mobile/ui/savings_account_application/SavingsAccountApplicationScreen.kt

```diff
@@ -0,0 +1,131 @@
+package org.mifos.mobile.ui.savings_account_application
+
+import android.widget.Toast
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import org.mifos.mobile.R
+import org.mifos.mobile.core.ui.component.EmptyDataView
+import org.mifos.mobile.core.ui.component.MifosProgressIndicator
+import org.mifos.mobile.core.ui.component.MifosTopBar
+import org.mifos.mobile.core.ui.component.NoInternet
+import org.mifos.mobile.core.ui.theme.MifosMobileTheme
+import org.mifos.mobile.models.accounts.savings.SavingsWithAssociations
+import org.mifos.mobile.utils.Network
+
+
+@Composable
+fun SavingsAccountApplicationScreen(
+    uiState: SavingsAccountApplicationUiState? = null,
+    savingsWithAssociations: SavingsWithAssociations? = null,
+    navigateBack: () -> Unit,
+    submit: (Int, Int) -> Unit,
+    retryConnection: () -> Unit,
+) {
+    var topBarTitleText by remember { mutableStateOf("") }
+    val context = LocalContext.current
+
+    Column {
+        MifosTopBar(
+            navigateBack = navigateBack,
+            title = { Text(text = topBarTitleText) }
+        )
+
+        when (uiState) {
+            is SavingsAccountApplicationUiState.Error -> {
+                ErrorComponent(
+                    retryConnection = retryConnection,
+                    errorMessage = uiState.errorMessage
+                )
+            }
+
+            is SavingsAccountApplicationUiState.Loading -> {
+                MifosProgressIndicator(modifier = Modifier.fillMaxSize())
+            }
+
+            is SavingsAccountApplicationUiState.ShowUserInterfaceSavingAccountApplication -> {
+                topBarTitleText = stringResource(id = R.string.apply_savings_account)
+                SavingsAccountApplicationContent(
+                    savingsAccountTemplate = uiState.template,
+                    submit = submit
+                )
+            }
+
+            is SavingsAccountApplicationUiState.ShowUserInterfaceSavingAccountUpdate -> {
```

Comment:
we can merge these two ShowUserInterfaceSavingAccountUpdate and ShowUserInterfaceSavingAccountApplication by maintaining the field in ViewModel 
enum class RequestType {
 NEW_APPLICATION, 
 UPDATE
 }
 
 So initially we keep the value in ViewModel and set the parameter here accordingly.